base/applications/osk/main.c

```diff
@@ -262,9 +262,18 @@ int OSK_DlgTimer(void)
     }
 
     /* Always redraw leds because it can be changed by the real keyboard) */
-    InvalidateRect(GetDlgItem(Globals.hMainWnd, IDC_LED_NUM), NULL, TRUE);
-    InvalidateRect(GetDlgItem(Globals.hMainWnd, IDC_LED_CAPS), NULL, TRUE);
-    InvalidateRect(GetDlgItem(Globals.hMainWnd, IDC_LED_SCROLL), NULL, TRUE);
+    if (GetAsyncKeyState(VK_NUMLOCK))
+    {
+        InvalidateRect(GetDlgItem(Globals.hMainWnd, IDC_LED_NUM), NULL, TRUE);
```

Comment:
Invalidate it every 10 ms like a madman when the state is `on`?
I would not call this an improvement